StellarApp/Controllers/NewsViewController.swift

```diff
@@ -36,37 +46,53 @@ class NewsViewController: UIViewController {
             }
         }
     }
-    
 }
-
-extension NewsViewController: UICollectionViewDataSource, UICollectionViewDelegate {
+extension NewsViewController: UICollectionViewDataSource {
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
         return articles.count
     }
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
         guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "NewsCell", for: indexPath) as? NewsCollectionViewCell else { return UICollectionViewCell() }
         let thisArticle = articles[indexPath.row]
-        cell.articleLabel.text = "Title: \(thisArticle.title)"
-        cell.articleDescription.text = "Description: \(thisArticle.description)"
+        cell.articleLabel.text = thisArticle.source.name
+        cell.articleDescription.text = thisArticle.title
         cell.backgroundColor = .clear
         let url = URL(fileURLWithPath: thisArticle.urlToImage)
         ImageHelper.fetchImageFromNetwork(urlString: thisArticle.urlToImage) { (error, data) in
             if let error = error {
                 print(error.errorMessage())
         }  else if let data = data {
                 cell.articleImage.layer.borderWidth = 5
-                cell.articleImage.layer.borderColor = UIColor.white.cgColor
                 cell.articleImage.layer.cornerRadius = 5
+                cell.articleImage.layer.borderColor = UIColor.clear.cgColor
                 cell.articleImage.image = data
+            
             }
         }
+        cell.backgroundColor = .lightGray
         return cell
     }
 }
-
-extension NewsViewController: UICollectionViewDelegateFlowLayout {
+extension NewsViewController: UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        guard  let url = URL(string: articles[indexPath.row].url) else {
+            return
+        }
+        
+        let safariVC = SFSafariViewController(url: url)
+        present(safariVC, animated: true, completion: nil)
+    }
     func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
-        return CGSize(width: self.view.frame.width * 0.8, height: 200)
+        return CGSize(width: view.bounds.width - 32, height: 300)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+        return 16.0
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return 16.0
```

Comment:
isn't these two does the same?

Reply:
No we do not need it Lee, You can delete it

Reply:
No they are not the same. The first one is for the inter item spacing and the second one is for marginal spacing